dev-client/src/model/color/colorDetection.ts

```diff
@@ -108,12 +108,17 @@ export const dominantColor = (pixels: RGBA[]): RGB => {
   });
 
   // Quantize.js performs median cut algorithm, and returns a palette of the "top 16" colors in the picture
-  var cmap = quantize(pixelArray, COLOR_COUNT);
-  if (cmap === false) {
+  var colorMap = quantize(pixelArray, COLOR_COUNT);
+  if (colorMap === false) {
     throw new Error('Unexpected color algorithm failure!');
   }
 
-  return cmap.palette()[0];
+  const colorsWithCounts = colorMap.vboxes.map(
+    vbox => [vbox.color, vbox.vbox.count()] as const,
+  );
+  const [dominantColor] = colorsWithCounts.sort(([, c1], [, c2]) => c2 - c1)[0];
+
+  return dominantColor;
```

Comment:
Could extract this into a testable function like `getDominantColor` or something. 

Also, out of curiosity, what's a vbox?

Reply:
i don't fully understand, the juicy details are explained in [this paper](http://leptonica.org/papers/mediancut.pdf), but vaguely speaking it's a range of pixel values with associated frequency count